src/printer.rs

```diff
@@ -960,19 +971,30 @@ pub enum TypeMetadata {
         name: String,
         adt_def: AdtDef,
     },
+    UnionType {
+        name: String,
+        adt_def: AdtDef,
+    },
+    ArrayType(stable_mir::ty::Ty, Option<stable_mir::ty::TyConst>),
+    PtrType(stable_mir::ty::Ty),
+    RefType(stable_mir::ty::Ty),
+    TupleType(Vec<stable_mir::ty::Ty>),
```

Comment:
```suggestion
    TupleType {
        types: Vec<stable_mir::ty::Ty>
    },
```

The json parser is giving me issues when this field is unnamed.

Reply:
Sure, we can change that (incl. golden testing).
The problem arises because if there is more than one argument to the constructor (e.g., `ArrayType`) the unnamed arg.s will be encoded as a JSON array.